app/models/agents/weather_agent.rb

```diff
@@ -100,14 +107,26 @@ def validate_options
     end
 
     def wunderground
-      Wunderground.new(interpolated['api_key']).forecast_for(location)['forecast']['simpleforecast']['forecastday'] if key_setup?
+      if key_setup?
+        forecast = Wunderground.new(interpolated['api_key'], language: language.upcase).forecast_for(location)
+
+        merged = Hash.new
+        forecast['forecast']['simpleforecast']['forecastday'].each{|daily| merged[daily['period']] = daily}
+
+        txt_forecast = Hash.new
+        merged.each_key do |day|
+            txt_forecast[day] = forecast['forecast']['txt_forecast']['forecastday'].find{|x| x['period'] == day}
+        end
+
+        merged.each_key{|day| merged[day] = merged[day].merge(txt_forecast[day])}
```

Comment:
All of the currently existing keys will remain the same, but a few other will be added from the `forecast.txt_forecast.forecastday`. So the format remains the same. See a [sample JSON here](http://api.wunderground.com/api/9ed080caa80b5d90/forecast/q/zmw:00000.13.12982.json). 

Furthermore it fixes a bug as well. Instead of using `which_day` as [array index](https://github.com/elvetemedve/huginn/blob/a0f19652084cb2ddc679018a7c0d1c79dfb9f1ac/app/models/agents/weather_agent.rb#L135), now the data of a day will be [read by the `period`](https://github.com/cantino/huginn/pull/1544/files#diff-b4d5bdeb4cc4b27e5cfe85ec715945dcR121) value. Currently if you set `which_day` to **1**, the second array item will be returned which has period equals to **2**, but should be **1**.